src/extensionsIntegrated/Phoenix/profile-menu.js

```diff
@@ -0,0 +1,229 @@
+                closePopup();
+            }
+        }
+    }
+
+    /**
+     * this function is to position the popup near the profile button
+     */
+    function positionPopup() {
+        const $profileButton = $("#user-profile-button");
+
+        if ($profileButton.length && $popup) {
+            const buttonPos = $profileButton.offset();
+            const popupWidth = $popup.outerWidth();
+            const windowWidth = $(window).width();
+
+            // pos above the profile button
+            let top = buttonPos.top - $popup.outerHeight() - 10;
+
+            // If popup would go off the right edge of the window, align right edge of popup with right edge of button
+            let left = Math.min(
+                buttonPos.left - popupWidth + $profileButton.outerWidth(),
+                windowWidth - popupWidth - 10
+            );
+
+            // never go off left edge
+            left = Math.max(10, left);
+
+            $popup.css({
+                top: top + "px",
+                left: left + "px"
+            });
+        }
+    }
+
+    /**
+     * Shows the sign-in popup when the user is not logged in
+     * @param {Object} loginData - Data to populate the template (optional)
+     */
+    function showLoginPopup(loginData) {
+        // If popup is already visible, just close it
+        if (isPopupVisible) {
+            closePopup();
+            return;
+        }
+
+        // Merge provided data with defaults
+        const templateData = $.extend({}, defaultLoginData, loginData || {});
+
+        // create the popup element
+        closePopup(); // close any existing popup first
+
+        // Render template with data
+        const renderedTemplate = Mustache.render(loginTemplate, templateData);
+        $popup = $(renderedTemplate);
+
+        $("body").append($popup);
+        isPopupVisible = true;
+
+        positionPopup();
+
+        // event handlers for buttons
+        $popup.find("#phoenix-signin-btn").on("click", function () {
+            _handleSignInBtnClick();
+            closePopup();
+        });
+
+        $popup.find("#phoenix-support-btn").on("click", function () {
+            _handleContactSupportBtnClick();
+            closePopup();
+        });
+
+        // Set up global click handler to close popup when clicking outside
+        // Delay attaching to avoid immediate closing
+        setTimeout(function () {
+            $(document).on("click.profilePopup", handleDocumentClick);
+        }, 0);
+    }
+
+    /**
+     * Shows the user profile popup when the user is logged in
+     * @param {Object} profileData - Data to populate the template (optional)
+     */
+    function showProfilePopup(profileData) {
+        // If popup is already visible, just close it
+        if (isPopupVisible) {
+            closePopup();
+            return;
+        }
+
+        // Merge provided data with defaults
+        const templateData = $.extend({}, defaultProfileData, profileData || {});
```

Comment:
see usages for `PopUpManager.addPopUp` . it handles some cases like
1. if another popup is open, for example editor context menu, opening this popup should dismiss it and vice versa.
2. Escape key handling.